beacon-chain/verification/blob_test.go

```diff
@@ -0,0 +1,660 @@
+package verification
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/db"
+	forkchoicetypes "github.com/prysmaticlabs/prysm/v4/beacon-chain/forkchoice/types"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/startup"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v4/testing/require"
+	"github.com/prysmaticlabs/prysm/v4/testing/util"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+)
+
+func TestBlobIndexInBounds(t *testing.T) {
+	ini := &Initializer{}
+	_, blobs := util.GenerateTestDenebBlockWithSidecar(t, [32]byte{}, 0, 1)
+	b := blobs[0]
+	// set Index to a value that is out of bounds
+	v := ini.NewBlobVerifier(b, GossipSidecarRequirements...)
+	require.NoError(t, v.BlobIndexInBounds())
+	require.Equal(t, true, v.results.executed(RequireBlobIndexInBounds))
+	require.NoError(t, v.results.result(RequireBlobIndexInBounds))
+
+	b.Index = fieldparams.MaxBlobsPerBlock
+	v = ini.NewBlobVerifier(b, GossipSidecarRequirements...)
+	require.ErrorIs(t, v.BlobIndexInBounds(), ErrBlobIndexInBounds)
+	require.Equal(t, true, v.results.executed(RequireBlobIndexInBounds))
+	require.NotNil(t, v.results.result(RequireBlobIndexInBounds))
+}
+
+func TestSlotBelowMaxDisparity(t *testing.T) {
+	now := time.Now()
+	// make genesis 1 slot in the past
+	genesis := now.Add(-12 * time.Second)
```

Comment:
is there a reason not to use `params.BeaconConfig().SecondsPerSlot` here?

Reply:
updated